packages/labs/task/src/task.ts

```diff
@@ -44,6 +44,7 @@ export interface TaskConfig<T extends ReadonlyArray<unknown>, R> {
   task: TaskFunction<T, R>;
   args?: ArgsFunction<T>;
   autoRun?: boolean;
+  throwErrors?: boolean;
```

Comment:
I would love to find a way to address this without a new config option. I think an expansion of options can be bad DX as users have to decide whether they should say the option or not, and what the default it.

I was thinking we could make taskComplete a getter, so we don't even create a promise that could have an uncaught error. The getter would have to create an already resolved promise if the task is complete.

Your comment about an error renderer is insightful! It's be great if we detected an error renderer and also suppressed uncaught errors. And maybe the same for the onError callback?

Reply:
What is the advantage of the getter? Is it just not having to create the promise in the first place? If so, I feel like the bloat of the extra checks around it might outweigh the memory savings if that's the only reason

Reply:
I like the onError and render.errror suggestion. Would remove the need for the config option

Reply:
The getter is so that we avoid even making the promise until it's been asked for, so by default we don't get any uncaught errors. Then if a user does access the promise, and don't handle the error, they'll get the warning if they don't have the error, but in that case they really should get the warning.

Reply:
Actually, I think with the on-demand promise creation, you won't need to look at error handlers, since the warning would only ever be made of the user gets a promise and doesn't handle it correctly.

Reply:
issues with the automatic error handling via `shouldCatch = !!this.onError || this.render.error` – it's difficult to determine whether the user has an error handler in the error message. We could do something like:

`this._hasErrorRenderer = this._hasErrorRenderer || !!renderer.error` in `render()`, but that requires:

1. The user to have called `render()` at least once with an `error` option before the task has completed
   - The user could be rendering their error renderer _after_ the task has completed in a conditional template
2. The user to continue rendering the `error` renderer because the logic is optimistic that there is an error handler to handle the following case:
   - ```js
      ${task.render({error: ... })}
      ${task.render({complete: ... })}
      ```

I'd also wager that the user having to handle catch on `taskComplete` is not great because it doesn't make sense that you'd have to catch on `taskComplete` as opposed to the task itself handling it.

WDYT of:

- Do not turn `taskComplete` into a getter
- No `throwErrors` and just catch every time and it's up to the user to use our intended API to handle errors at all time

Reply:
I think making `taskComplete` a getter solves this. There's no need to look for an error handler in render. The logic should be simple: if a user called `.taskCompete`, reject that promise on error, if not, there's no promise to reject and no unhandled rejection.

We already do reject the taskComplete promise, so this wouldn't even be a breaking change. If we don't reject we have to at least resolve it, so we don't leave it hanging, but I think that's a bit weird.

Reply:
took a while to untangle the logic in my head, but tests should be robust. PTAL

Reply:
updated main comment with a gotcha, but it's just how Promises work